x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackTemplateRegistry.java

```diff
@@ -249,10 +262,97 @@ protected List<LifecyclePolicy> getLifecyclePolicies() {
             }
         }
         COMPONENT_TEMPLATE_CONFIGS = Map.copyOf(componentTemplates);
+
+        final Map<String, ComponentTemplate> logsdbComponentTemplates = new HashMap<>();
```

Comment:
I only think we need to load different `ComponentTemplate` instances for logs mappings and settings? These will replace the non logsdb mappings and settings. No need to to load all component templates twice, the logsdb setting is non dynamic, so at node startup we can determine what needs to be loaded.

Reply:
```
this.logsIndexModeTemplateEnabled = CLUSTER_LOGSDB_ENABLED.get(nodeSettings);
```

`nodeSettings` are only available when the construction of `StackTemplateRegistry` happens...

Reply:
Ah right, I see. The issue here is that the component templates map is loaded when the `StackTemplateRegistry` class is loaded. At that time we don't have access to node settings. If the map for component template was initialized in the constructor we can have access to it (adding a `Settings` parameter is supported).  

Then let's leave this as is.